tgui/packages/tgui/interfaces/Manufacturer/index.tsx

```diff
@@ -132,6 +133,19 @@ export const Manufacturer = () => {
   const manudriveName = manudrive?.name ?? '';
   const manudriveLimit = manudrive?.limit;
 
+  // Stores a ref copy of the resource+producibility data which is updated ONLY when the contents of the manufacturer are changed.
+  // This helps make sure that the buttons can actually be memoized, as they wouldn't otherwise.
+  const resource_data_ref = useRef(resource_data);
+  const producibility_data_ref = useRef(producibility_data);
+
+  // Checking for a change in resource data via stringified vars is effectively checking if contents change
+  if (
+    JSON.stringify(resource_data) !== JSON.stringify(resource_data_ref.current)
+  ) {
+    producibility_data_ref.current = producibility_data;
+    resource_data_ref.current = resource_data;
+  }
```

Comment:
So this is invalidating a producibility_data/resource_data cache whenever resource_data changes? What about if producibility_data changes? Should this not be simply worrying about producibility_data? Iterating over a short list that is keyed by their byondRef will avoid most of the rendering cost of the materials list. Not sure much benefit to caching resource_data.